pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -98,13 +105,131 @@ type DockercfgController struct {
 
 	serviceAccountCache      MutationCache
 	serviceAccountController *framework.Controller
+	secretController         *framework.Controller
 
 	queue workqueue.RateLimitingInterface
 
 	// syncHandler does the work. It's factored out for unit testing
 	syncHandler func(serviceKey string) error
 }
 
+func (e *DockercfgController) secretTokenAvailable(_, newObj interface{}) {
```

Comment:
This can happen on an `Add` instead of on a `Update` if your controller restarts after having been off.  See https://github.com/kubernetes/kubernetes/pull/32778